### PR TITLE
feat(repo): add a counter field

### PR DESCRIPTION
**Pull request by kneal**
PR Includes the following:

* adds a `counter` variable to the repo database struct
* adds a `counter` variable to the repo library struct

Related to: https://github.com/go-vela/community/issues/137

### CODE REVIEW

File: library/repo.go [hunk 1]
```diff
@@ -584,6 +611,7 @@ func (r *Repo) String() string {
   Org: %s,
   Private: %t,
   Timeout: %d,
+	Counter: %d,
```
Comment (jbrockopp) [inline library/repo.go:614]: Could we fix the spacing here so it's inline with the other fields?

File: library/repo_test.go [hunk 2]
```diff
@@ -290,6 +291,7 @@ func TestLibrary_Repo_String(t *testing.T) {
   Org: %s,
   Private: %t,
   Timeout: %d,
+	Counter: %d,
```
Comment (jbrockopp) [inline library/repo_test.go:294]: Could we fix the spacing here so it's inline with the other fields?

**Review by jbrockopp — CHANGES_REQUESTED** [review]

File: library/repo.go (hunk 1 above)
Comment (kneal) [reply library/repo.go:614]: Oh weird, my editor rendered it fine. I'll take a look

File: library/repo_test.go (hunk 2 above)
Comment (kneal) [reply library/repo_test.go:294]: Oh weird, my editor rendered it fine. I'll take a look

Comment (kneal) [reply library/repo_test.go]: https://github.com/go-vela/types/pull/176/commits/4e63050571a1d705ee5efd1a816b6b17bd2de2f4

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by KellyMerrick — APPROVED** [review]
lgtm

**Pull request merged**